examples/vmware-cloud-director-machinedeployment.yaml

```diff
@@ -0,0 +1,87 @@
+    metadata:
+      labels:
+        foo: bar
+    spec:
+      providerSpec:
+        value:
+          sshPublicKeys:
+            - "<< YOUR_PUBLIC_KEY >>"
+          cloudProvider: "vmware-cloud-director"
+          cloudProviderSpec:
+            # Can also be set via the env var 'VCD_USER' on the machine-controller
+            username: "<< VCD_USER >>"
+            # Can also be set via the env var 'VCD_URL' on the machine-controller
+            # example: 'https://your-vcloud-director:8443'. '/api' gets appended automatically
+            url: "<< VCD_URL >>"
+            # Can also be set via the env var 'VCD_PASSWORD' on the machine-controller
+            password:
+              secretKeyRef:
+                namespace: kube-system
+                name: machine-controller-vcloud-director
+                key: password
+            # Can also be set via the env var 'VCD_ORG' on the machine-controller
+            organization: "<< VCD_ORG >>"
+            # Can also be set via the env var 'VCD_VDC' on the machine-controller
+            vdc: "<< VCD_VDC >>"
+            # Can also be set via the env var 'VCD_ALLOW_UNVERIFIED_SSL' on the machine-controller
+            allowInsecure: false
+            # vApp to associate the VM with. This should be created before the machine is created
+            vapp: "<< VCD_VAPP >>"
+            # Name of catalog where the VM template is located
+            catalog: "<< VCD_CATALOG >>"
+            # Name of OS template to be used for the VM
+            template: "<< VCD_TEMPLATE >>"
+            # Direct or routed network that can be used for the VM
+            network: "<< VCD_NETWORK >>"
+            ipAllocationMode: "DHCP"
+            cpus: 2
+            cpuCores: 1
+            memoryMB: 2048
+            # Optional: Resize the root disk to this size. Must be bigger than the existing size
+            # Default is to leave the disk at the same size as the template
+            diskSizeGB: 10
+            diskBusType: "paravirtual"
+            diskIOPS: 0
+            storageProfile: "*"
+          operatingSystem: "ubuntu"
+          operatingSystemSpec:
+            distUpgradeOnBoot: false
+            disableAutoUpdate: true
+            # 'rhelSubscriptionManagerUser' is only used for rhel os and can be set via env var `RHEL_SUBSCRIPTION_MANAGER_USER`
+            rhelSubscriptionManagerUser: "<< RHEL_SUBSCRIPTION_MANAGER_USER >>"
+            # 'rhelSubscriptionManagerPassword' is only used for rhel os and can be set via env var `RHEL_SUBSCRIPTION_MANAGER_PASSWORD`
+            rhelSubscriptionManagerPassword: "<< RHEL_SUBSCRIPTION_MANAGER_PASSWORD >>"
+            # 'rhsmOfflineToken' if it was provided red hat systems subscriptions will be removed upon machines deletions, and if wasn't
+            # provided the rhsm will be disabled and any created subscription won't be removed automatically
+            rhsmOfflineToken: "<< REDHAT_SUBSCRIPTIONS_OFFLINE_TOKEN >>"
```

Comment:
ubuntu's Config structure doesn't have those fields

https://github.com/kubermatic/machine-controller/blob/ebbd117869da8a35867e74193510ef6241adb4e1/pkg/userdata/ubuntu/ubuntu.go#L25-L28

Reply:
True but does it matter? I think we add all of this by default in all of our examples and testdata so that switching between different OS is easier. It will get ignored since this is not strictly unmarshalled.

Reply:
Yes, let's not confuse people with non-existing fields.

Reply:
Updated PTAL 👍